### PR TITLE
fix: fix type assertion panic

### LINKED ISSUE #91: SET @@session.autocommit = OFF lead server panic for interface type assertion.
- [x] I have searched the [issues](https://github.com/cectc/dbpack/issues) of this repository and believe that this is not a duplicate.

**Describe the bug**
`SET @@session.autocommit = OFF` lead server panic for interface type assertion.

**To Reproduce**
1. run dbpack from command line
2. run python-samples order demo

**Expected behavior**
`SET @@session.autocommit = OFF` execute well, not panic the server.

**Screenshots**

image

**Additional context**

### PR DESCRIPTION
**Pull request by fatelei**
ref: https://github.com/cectc/dbpack/issues/<issueID>

### Ⅰ. Describe what this PR did

fix #91 

### Ⅱ. Does this pull request fix one issue?

#91

### CODE REVIEW

**Review by dk-lockdown — APPROVED** [review]
LGTM

File: pkg/executor/single_db.go [hunk 1]
```diff
@@ -139,23 +139,25 @@ func (executor *SingleDBExecutor) ExecutorComQuery(ctx context.Context, sql stri
...
 	case *ast.SetStmt:
-		if len(stmt.Variables) == 1 &&
-			strings.EqualFold(stmt.Variables[0].Name, "autocommit") &&
-			stmt.Variables[0].Value.(*driver.ValueExpr).GetValue() == int64(0) {
-			tx, result, err = db.Begin(ctx)
-			if err != nil {
-				return nil, 0, err
+		if len(stmt.Variables) == 1 && strings.EqualFold(stmt.Variables[0].Name, "autocommit") {
+			valueExpr, ok := stmt.Variables[0].Value.(*driver.ValueExpr)
+			if ok && valueExpr.GetValue() == int64(0) {
```
Comment (dk-lockdown) [inline pkg/executor/single_db.go]: `SET @@session.autocommit=OFF` should begin transaction too

File: pkg/executor/single_db.go [hunk 2]
```diff
@@ -139,23 +139,35 @@ func (executor *SingleDBExecutor) ExecutorComQuery(ctx context.Context, sql stri
...
 	switch stmt := queryStmt.(type) {
 	case *ast.SetStmt:
-		if len(stmt.Variables) == 1 &&
-			strings.EqualFold(stmt.Variables[0].Name, "autocommit") &&
-			stmt.Variables[0].Value.(*driver.ValueExpr).GetValue() == int64(0) {
-			tx, result, err = db.Begin(ctx)
-			if err != nil {
-				return nil, 0, err
+		if len(stmt.Variables) == 1 && strings.EqualFold(stmt.Variables[0].Name, "autocommit") {
+			valueExpr, ok := stmt.Variables[0].Value.(*driver.ValueExpr)
```
Comment (dk-lockdown) [inline pkg/executor/single_db.go]: suggest use switch case instead of if else

File: pkg/executor/single_db.go [hunk 3]
```diff
@@ -139,23 +139,35 @@ func (executor *SingleDBExecutor) ExecutorComQuery(ctx context.Context, sql stri
...
+			canStartTransaction := false
+			if ok && valueExpr.GetValue() == int64(0) {
+				canStartTransaction = true
+			} else {
+				columnNameExpr, ok := stmt.Variables[0].Value.(*ast.ColumnNameExpr)
+				if ok && strings.ToLower(columnNameExpr.Name.String()) == "off" {
```
Comment (dk-lockdown) [inline pkg/executor/single_db.go]: suggest use `strings.EqualFold() `

File: pkg/executor/single_db.go [hunk 4]
```diff
@@ -139,23 +139,36 @@ func (executor *SingleDBExecutor) ExecutorComQuery(ctx context.Context, sql stri
...
-			stmt.Variables[0].Value.(*driver.ValueExpr).GetValue() == int64(0) {
-			tx, result, err = db.Begin(ctx)
-			if err != nil {
-				return nil, 0, err
+		if len(stmt.Variables) == 1 && strings.EqualFold(stmt.Variables[0].Name, "autocommit") {
+			canStartTransaction := false
+			switch exprType := stmt.Variables[0].Value.(type) {
+			case *driver.ValueExpr:
+				if exprType.GetValue() == int64(0) {
+					canStartTransaction = true
```
Comment (dk-lockdown) [inline pkg/executor/single_db.go]: pls rename to  shouldStartTransaction

**Comment by adofsauron** [comment]
LGTM

File: pkg/executor/single_db.go [hunk 5]
```diff
@@ -139,23 +139,36 @@ func (executor *SingleDBExecutor) ExecutorComQuery(ctx context.Context, sql stri
 	db = resource.GetDBManager().GetDB(executor.dataSource)
 	switch stmt := queryStmt.(type) {
 	case *ast.SetStmt:
-		if len(stmt.Variables) == 1 &&
-			strings.EqualFold(stmt.Variables[0].Name, "autocommit") &&
-			stmt.Variables[0].Value.(*driver.ValueExpr).GetValue() == int64(0) {
-			tx, result, err = db.Begin(ctx)
-			if err != nil {
-				return nil, 0, err
+		if len(stmt.Variables) == 1 && strings.EqualFold(stmt.Variables[0].Name, "autocommit") {
```
Comment (bohehe) [inline pkg/executor/single_db.go]: can use const instead of magic string "auto commit"

File: pkg/executor/single_db.go [hunk 6]
```diff
@@ -139,23 +139,36 @@ func (executor *SingleDBExecutor) ExecutorComQuery(ctx context.Context, sql stri
...
+			shouldStartTransaction := false
+			switch exprType := stmt.Variables[0].Value.(type) {
+			case *driver.ValueExpr:
+				if exprType.GetValue() == int64(0) {
+					shouldStartTransaction = true
+				}
+			case *ast.ColumnNameExpr:
+				if strings.EqualFold(exprType.Name.String(), "off") {
```
Comment (bohehe) [inline pkg/executor/single_db.go]: can use const instead of magic string "off"

**Review by bohehe — APPROVED** [review]
lgtm

**Pull request merged**